remind_mfa/plastics/plastics_mfa_system.py

```diff
@@ -196,77 +167,94 @@ def compute_flows(self):
             "virgin_material_shares": self.get_new_array(dim_letters=("t", "e", "r", "m")),
             "captured_2_virginccu_by_mat": self.get_new_array(dim_letters=("t", "e", "r", "m")),
             "ratio_nonc_to_c": self.get_new_array(dim_letters=("m",)),
-            "final_2_fabrication": self.get_new_array(dim_letters=("t", "e", "m")),
         }
 
-        # non-C atmosphere & captured has no meaning & is equivalent to sysenv
-        material_element_split = prm["material_shares_in_goods"] * prm["carbon_content_materials"]
-        good_split = stk["in_use"].inflow.sum_over(("e", "m")).get_shares_over("g")
-        good_split_eol = stk["in_use"].outflow.sum_over(("e", "m")).get_shares_over("g")
-        material_element_split_noGood = (
-            stk["in_use"].inflow.sum_over(("g")).get_shares_over(("e", "m"))
+        split_use = stk["in_use"].inflow.get_shares_over(("g", "e", "m"))
+        extrapolate_trade(
+            self.split_trade_by_share(historic_trade["final_his"], split_use),
+            self.trade_set["final"],
+            stk["in_use"].inflow,
+            "imports",
+            balance_to="hmean",
         )
 
-        flw["primary_market => primary_imports"][...] = (
-            trd["primary"].imports * material_element_split_noGood
+        flw["good_market => use"][...] = trd["final"].imports
+        flw["fabrication => good_market"][...] = trd["final"].exports
+        flw["sysenv => good_market"][...] = flw["good_market => use"][...]
```

Comment:
The `[...]` on the right-hand side of an assignment doesn't do anything if you have it on the left-hand side.

Reply:
there's quite some occurences of this, best to check the whole compute function...